boudicca.base/enricher/src/main/kotlin/base/boudicca/enricher/service/osm/OsmEnricher.kt

```diff
@@ -0,0 +1,193 @@
+            property = SemanticKeys.LOCATION_COORDINATES_LON_PROPERTY,
+            value = center["lon"].asText(),
+            variants = sourceVariant
+        ).withProperty(
+            property = SemanticKeys.SOURCES_PROPERTY,
+            value = currentSources + "Source:OSM:" + json["osm3s"]["copyright"].asText()
+        )
+
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_OSM_ID_PROPERTY,
+            osmId
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_NAME_PROPERTY,
+            tags["name"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_ADDRESS_PROPERTY,
+            buildAddress(tags)
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_CITY_PROPERTY,
+            tags["addr:city"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_DESCRIPTION_PROPERTY,
+            tags["amenity"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_CONTACT_EMAIL_PROPERTY,
+            tags["contact:email"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_CONTACT_PHONE_PROPERTY,
+            tags["contact:phone"]?.asText()
+        )
+        try {
+            builder.updatePropertyIfEmpty(
+                event,
+                SemanticKeys.LOCATION_URL_PROPERTY,
+                UrlUtils.parse(tags["contacts:website"]?.asText())
+            )
+        } catch (_: IllegalArgumentException) {
+            logger.info("Could not map ${tags["contacts:website"].asText()} to URI")
+        }
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_WIKIPEDIA_PROPERTY,
+            tags["wikipedia"]?.asText()
+        )
+        builder.updatePropertyIfEmpty(
+            event,
+            SemanticKeys.LOCATION_WIKIDATA_PROPERTY,
+            tags["wikidata"]?.asText()
+        )
+
+        return builder.build()
+    }
+
+    private fun getOrSearchOsmId(event: StructuredEvent): String? {
+        // TODO maybe think about taking all osmIds and not only the first :)
+        val osmIds = event.getProperty(SemanticKeys.LOCATION_OSM_ID_PROPERTY)
+        val osmId = if (osmIds.isNotEmpty()) {
+            osmIds.first().second
+        } else {
+            searchOsmId(event)
+        }
+        return osmId
+    }
+
+    private fun searchOsmId(event: StructuredEvent): String? {
+        val eventName = event
+            .getProperty(SemanticKeys.LOCATION_NAME_PROPERTY)
+            .map(Pair<Key, String>::second)
+            .joinToString(separator = ",") { it.encodeURL() }
+
+        val query = """https://nominatim.openstreetmap.org/search?q=$eventName&format=json&limit=1"""
```

Comment:
we cannot merge it like this. the site specifies a heavy "max 1 request per second" policy here: https://operations.osmfoundation.org/policies/nominatim/

we need to honor this

Reply:
it also may be that we are not allowed to use it at all, they say as unacceptable use: "Scraping of details The details page is there for debugging only and may not be downloaded automatically."

also they say "As a general rule, bulk geocoding of larger amounts of data is not encouraged. If you have regular geocoding tasks, please, look into alternatives below."